### PR TITLE
Update Rust dependencies to latest versions

### PR DESCRIPTION
**Pull request by massand**
Bumping major versions of dependencies after the recent release of tokio 1.0

### CODE REVIEW

**Comment by arsing** [comment]
Lockfile still has tokio 0.2. You need to update whatever's using it too. (Search for `"tokio 0`)

**Comment by massand** [comment]
> Lockfile still has tokio 0.2. You need to update whatever's using it too. (Search for `"tokio 0`)

Sorry I forgot to call this out in the PR description - I had lots of issues converting the `aziot-key-openssl-engine-shared-test` crate due to changes in some dependency crates in how they used to wrap `openssl` crate types, but now got rid of those wrappers (e.g. `HandshakeError`). We can independently do this (since it is a test only crate) and the last tokio 0.2 dependency will dissappear.

**Comment by massand** [comment]
@arsing - Question about the failing check: have you hit licensing issues with `tracing-futures` crate in the past? It doesn't seem like a new dependency - maybe a new version?

**Comment by arsing** [comment]
Take https://github.com/arsing/iot-identity-service/commits/tokioup . Everything is fixed in that.

**Comment by arsing** [comment]
Rename the `tokio02` feature too please, to `tokio1`.

**Review by arsing — APPROVED** [review]

File: http-common/src/connector.rs [hunk 1]
```diff
@@ -398,13 +395,32 @@ impl tokio::io::AsyncWrite for AsyncStream {
...
+    fn is_write_vectored(&self) -> bool {
+        match self {
+            AsyncStream::Tcp(inner) => inner.is_write_vectored(),
+            AsyncStream::Unix(inner) => inner.is_write_vectored(),
+        }
+    }
 }
 
-#[cfg(feature = "tokio02")]
+#[cfg(feature = "tokio1")]
```
Comment (arsing) [inline http-common/src/connector.rs:418]: You forgot to revert this. I'll send out a fix.

**Pull request merged**